tests/test_api.py

```diff
@@ -370,97 +374,94 @@ def test_node_dependencies(self, client, data_model_jsonld, source_node, return_
                 assert "TissueStatus" in response_dt
 
 @pytest.mark.schematic_api
-class TestManifestOperation:
-
+class TestManifestOperation():
     def ifExcelExists(self, response, file_name):
-        # return one excel file
+        # check if excel file gets returned as an attachment
         d = response.headers['content-disposition']
         fname = re.findall("filename=(.+)", d)[0]
         assert fname == file_name
     
-    def ifGoogleSheetExists(self, response_dt):
+    def ifGoogleSheetExists(self, response):
+        # check if response data contains google sheet
+        response_dt = json.loads(response.data)
         for i in response_dt: 
             assert i.startswith("https://docs.google.com/")
-    def ifPandasDataframe(self, response_dt):
+    def ifPandasDataframe(self, response):
+        # check if response data contains pandas dataframe
+        response_dt = json.loads(response.data)
         for i in response_dt:
             df = pd.read_json(i)
             assert isinstance(df, pd.DataFrame)
 
-
-    #@pytest.mark.parametrize("output_format", [None, "excel", "google_sheet", "dataframe (only if getting existing manifests)"])
-    @pytest.mark.parametrize("output_format", ["excel"])
-    @pytest.mark.parametrize("data_type", ["Biospecimen", "Patient", "all manifests", ["Biospecimen", "Patient"]])
-    def test_generate_existing_manifest(self, client, data_model_jsonld, data_type, output_format, caplog):
-        # set dataset
-        if data_type == "Patient":
-            dataset_id = ["syn42171373"] #Mock Patient Manifest folder on synapse
-        elif data_type == "Biospecimen":
-            dataset_id = ["syn42171508"] #Mock biospecimen manifest folder
-        elif data_type == ["Biospecimen", "Patient"]:
-            dataset_id = ["syn42171508", "syn42171373"]
-        else: 
-            dataset_id = None #if "all manifests", dataset id is None
-
+    def ifOutputCorrect(self, response, output_format, data_type, multiple_data_types=False):
+        # check if the correct output format gets generated
+        # if there are multiple data types, a warning should be triggered
+        if output_format == "excel":
+            self.ifExcelExists(response, f"Example.{data_type}.manifest.xlsx")
+            if multiple_data_types: 
+                with pytest.warns(UserWarning):
+                    warnings.warn(f"Currently we do not support returning multiple files as Excel format at once. Only Example.{data_type}.manifest would get returned", UserWarning)
```

Comment:
Maybe I'm missing something. When you use `pytest.raises()` or `pytest.warns()`, you usually call the code that you're trying to test with the expectation that it'll raise an exception or warning, respectively. Here, you are hard-coding a warning in the test, so this `pytest.warns()` will always pass. This begs the question why this is even here.